modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -1135,6 +1139,208 @@ public void testDeleteOrganizationsForPagination() {
+            dependsOnMethods = "createOrganizationsForMetaAttributesPaginationTests")
+    public void testAddMetaAttributesToOrganizations() {
+
+        // Sorted in the order.
+        String[] attributes =
+                {"1", "2", "3", ":", "@", "A", "B", "C", "LMN", "PQR", "STU", "a", "b", "c", "fg", "jKL", "mNo", "x",
+                        "y", "z"};
+        metaAttributes = new ArrayList<>();
+
+        String organizationId = organizationForMetaAttributes.get(ORGANIZATION_ID);
+        assertNotNull(organizationId, "Organization ID should not be null");
+
+        for (String attribute : attributes) {
+            String requestBody =
+                    "[{\"operation\":\"ADD\",\"path\":\"/attributes/" + attribute + "\",\"value\":\"value-" +
+                            attribute + "\"}]";
+
+            String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + "/" + organizationId;
+            Response response = getResponseOfPatch(endpointURL, requestBody);
+
+            validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+            // Save the attributes to the instance variable.
+            Map<String, String> attrMap = new HashMap<>();
+            attrMap.put(ORGANIZATION_ID_ATTRIBUTE, organizationId);
+            attrMap.put(ORGANIZATION_META_ATTRIBUTE_ATTRIBUTE, attribute);
+            metaAttributes.add(attrMap);
+        }
+
+        int expectedAttributesSize = attributes.length;
+        Assert.assertEquals(metaAttributes.size(), expectedAttributesSize,
+                String.format("Expected %d meta attributes, but found %d.", expectedAttributesSize,
+                        metaAttributes.size()));
+    }
+
+    @DataProvider(name = "metaAttributesLimitValidationDataProvider")
+    public Object[][] metaAttributesLimitValidationDataProvider() {
+
+        return new Object[][]{
+                {1}, {2}, {3}, {5}, {10}, {13}
+        };
+    }
+
+    @Test(groups = "organizationMetaAttributesPaginationTests",
+            dependsOnMethods = "testAddMetaAttributesToOrganizations",
+            dataProvider = "metaAttributesLimitValidationDataProvider")
+    public void testGetPaginatedMetaAttributesWithLimit(int limit) {
+
+        String endpointURL =
+                ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_META_ATTRIBUTES_API_PATH + QUESTION_MARK +
+                        LIMIT_QUERY_PARAM + EQUAL + limit + AMPERSAND + RECURSIVE_QUERY_PARAM + EQUAL + false;
+        Response response = getResponseOfGetWithOAuth2(endpointURL, m2mToken);
+
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+        List<String> attributes = response.jsonPath().getList(ORGANIZATION_MULTIPLE_META_ATTRIBUTE_ATTRIBUTES);
+        Assert.assertEquals(attributes.size(), limit);
+
+        // Validate the order of the returned attributes.
+        for (int i = 0; i < limit; i++) {
+            Assert.assertEquals(attributes.get(i), metaAttributes.get(i).get(ORGANIZATION_META_ATTRIBUTE_ATTRIBUTE));
```

Comment:
Here we can directly use `metaAttributes.get(i)` if a string array list is used.

Reply:
Addressed with https://github.com/wso2/product-is/pull/21054/commits/7f95c8efe6686a47acfec90aa0973af6e6e3bcc7